runtime/vm/ContinuationHelpers.cpp

```diff
@@ -132,11 +137,14 @@ enterContinuation(J9VMThread *currentThread, j9object_t continuationObject)
 	Assert_VM_notNull(continuation);
 
 	/* let GC know we are mounting, so they don't need to scan us, or if there is already ongoing scan wait till it's complete. */
-	synchronizeWithConcurrentGCScan(currentThread, continuation);
+	currentThread->pendingMountContinuation = continuation;
```

Comment:
since pending continuation is really only interesting for GC, let's move setting/reseting its value inside synchronizeWithConcurrentGCScan (that is mostly GC centric)
probably next to release/acquire VM access

Reply:
wherever you end up setting/reseting pendingContinuation try adding random sleep before and after.
the point is to create timing whole between 2 related by physically disjoint pieces of info 
- pending continuation in carrier thread and mounted info in continuation 
- pending continuation being set and current continuation not set (or vice versa, depending of the order of operations)